cdap-kafka/src/main/java/co/cask/cdap/kafka/run/KafkaServerMain.java

```diff
@@ -136,29 +142,24 @@ public void destroy() {
     // Nothing to do
   }
 
-  private Properties generateKafkaConfig(int brokerId, String zkConnectStr, String hostname, int port,
-                                         int numPartitions, int replicationFactor, String logDir) {
-    Preconditions.checkState(port > 0, "Port number is invalid.");
-    Preconditions.checkState(numPartitions > 0, "Num partitions should be greater than zero.");
+  private int tryParse(String val) {
```

Comment:
so are you saying just call `parseInt` and then it will simply throw the exception?